### DISCUSSION TITLE
Parcel incorrectly rewrites media queries

### DISCUSSION THREAD

**Discussion by MrWalt** [parcel-bundler/parcel#8985] [Q&A]
```
 "devDependencies": {
    "@parcel/transformer-sass": "^2.8.3",
    "parcel": "^2.8.3"
  }
```

Those are all of the dependencies I have in my current test file. Upon writing a media query with sass, parcel rewrites the output and messes up the query.

Sass file:
```
h1 {
  color: red;
}

@media only screen and (max-width: 40em) {
  h1 {
    color: blue;
  }
}

```
CSS output: 
```
h1 {
  color: red;
}

@media only screen and (width <= 40em) {
  h1 {
    color: #00f;
  }
}

/*# sourceMappingURL=index.b1276d39.css.map */
```

As you can see in the CSS output file, the media query is different and this completely ruins said media query. Any enlightenment is greatly appreciated.

**Comment by devongovett** [comment]
This is equivalent syntax in modern browsers. You need to set a browserslist if you want the output to target older browsers.

**Reply by MrWalt** [reply]
Im having some trouble understanding, as I did not need to this before on my project from a month ago. Can you explain a bit more?

**Reply by leebert** [reply]
I encountered the same problem just this week after updating Parcel. Setting the **browserslist** field in **package.json** is having no effect. Any advice would be greatly appreciated as it has completely stopped our development efforts with Parcel.

**Reply by mischnic** [reply]
Using Parcel 2.9.1:
- with `"browserslist": "Chrome >= 110"`, I get `width <=` and
- with `"browserslist": "Chrome >= 100"`, I get `max-width:`

If this doesn't happen for you, please share a small reproduction

**Reply by MrWalt** [reply]
Using Parcel 2.9.1.

I have followed as you instruct, however I cannot seem to get it to work as you say,
In my package.json:

```
{
  "name": "test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "start": "parcel index.html"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@parcel/transformer-sass": "^2.9.1",
    "parcel": "^2.9.1"
  },
  "browserslist": "Chrome >= 100"
}
```

- I've also tried with a .browserslistrc file to no success. The outcome of **max-width** is still **width <=**.

This was a clean installment of Parcel on a test project created just to try this out. 
`npm i parcel@latest` then
`npm i @parcel/transformer-sass@latest`, nothing was modified other than including `"browserslist": "Chrome >= 100"` in package.json.
Any other advice would be greatly appreciated.

**Reply by mischnic** [reply]
I think the problem here is that Parcel in development mode (`parcel index.html`)  by default uses modern engines 
https://github.com/parcel-bundler/parcel/blob/2a73cd2e7cf10d3242bbe4c15f87000f909cf4a4/packages/core/core/src/requests/TargetRequest.js#L82-L87
so in this case `yarn start` ignores the browserslist setting.


https://parceljs.org/features/development/#development-target:

> If you don't have any explicit targets defined, and only have a `browserslist` in your package.json, you can use the implicit default target with `--target default`. This will result in your source code being transpiled just as it would be in production.

**Reply by MrWalt** [reply]
Could you please show me how I can do this? I am a bit new and find it confusing, apologies.

**Reply by mischnic** [reply]
```json
  "scripts": {
    "start": "parcel index.html --target default"
  },
```

**Reply by MrWalt** [reply]
Thank you for the swift response but alas I have already tried it exactly like this and I have received a build error:
```
npm start

> test@1.0.0 start
> parcel index.html --target default

Server running at http://localhost:1234
🚨 Build failed.

@parcel/core: Could not find target with name "default"
```

**Reply by mischnic** [reply]
Do you have a `targets` field in package.json? Then you need to use one of those

**Reply by MrWalt** [reply]
I do not have a targets field in my package.json. Once again my apologies but what exactly do I put in targets?

**Reply by mischnic** [reply]
[media-queries.zip](https://github.com/parcel-bundler/parcel/files/11643059/media-queries.zip)

**Reply by leebert** [reply]
I updated to the latest version of Parcel (2.9.1) and my build is now working as expected. If it helps, here is the relevant part of my package.json:

```
  "scripts": {
    "start": "parcel './src/*.html' --target default",
    "releaseBuild": "npx parcel build './src/*.html'"
  },
  "browserslist": "> 0.5%, last 2 versions, not dead",
  "targets": {
    "default": {
      "publicUrl": "/"
    }
  },
```

_(note: I had to create a "dummy" default target to get the dev build to work correctly. This is probably the incorrect way to do it, but I'm still a newbie to all of this.)_

**Reply by MrWalt** [reply]
I've followed your advice and I have found what I am looking for, thank you very much for all of your suggestions, here is a **TLDR** for anyone else having the same issue:

- In my package.json file I have changed the following:

1. Changed the `"main": "index.js"` to `"default": "index.html"`
2. Added `"browserslist": "> 0.5%, last 2 versions, not dead"` or `"browserslist": "Chrome >= 100"`
3. Changed my 
```
 "scripts": {
    "start": "parcel index.html",
    "build": "parcel build index.html"
  },
``` 
to 
 ```
"scripts": {
    "start": "parcel index.html --target default",
    "build": "parcel build index.html"
  },
```
My media queries are working perfectly fine now, once again thank you.

**Reply by devongovett** [reply]
One question: what browser version are you developing in? As mentioned above, the default targets for development are the latest version of each browser, all of which support this. Are you on an older version intentionally?

**Reply by leebert** [reply]
For me, the `max-width:` media queries that were transpiled to `width <=`  were working correctly in Brave (Chromium 113). However, on Safari 16.3 in **Responsive Design Mode** they were not. Additionally, when connecting my phone (iOS 16.3.1) to my localhost server to test on-device none of the mobile browsers (Brave, Firefox, or Safari) were recognizing the media queries either. If I then used **Web Inspector** to change the media queries back to their original format everything started working as expected.

**Reply by devongovett** [reply]
I guess because 16.4 is the latest, not 16.3. Maybe we should change the default to last _major_ version?

**Reply by leebert** [reply]
That is a good question. Might that change lead to unintended side effects with existing projects that use Parcel, or subvert the expectations of experienced developers in terms of intended behavior for bundlers? Even though it was an initially frustrating experience for me when my code suddenly stopped working the end result was that I deepened my understanding of how bundlers work and gained additional web development experience points, which is a good thing.

**Reply by MrWalt** [reply]
I am using the latest version of Chrome and they work just fine there, however as leebert said `width <=` does not work for Safari 16.3 which is the reason I need `max-width:`. Its not a good practice to completely drop support for a browser that is so little behind from its latest.
